lib/src/container/store.rs

```diff
@@ -286,6 +303,14 @@ impl ImageImporter {
         self.prepare_internal(false).await
     }
 
+    /// Create a channel receiver that will get notifications for layer fetches.
+    pub fn request_progress(&mut self) -> Receiver<ImportProgress> {
+        assert!(self.layer_progress.is_none());
```

Comment:
It sounds like we wanted a spmc channel here instead, but tokio may not have one at hand.

Reply:
I think in practice any "multiple viewers" aspect will need to be handled at a higher level - rpm-ostree has quite complex infrastructure already for this.